### PR TITLE
feat: user v3 api update 

### LINKED ISSUE #7898: Command Side for Users with User Schemas (Updates)
As a ZITADEL Administrator I want to be able to update users and their states that apply to a predefined user schema, so I am able to ensure that my user data is consistent and applies to my standards.

## Acceptance Criteria
- [x] Feature is behind a feature flag
- [x] **UpdateUser**: changes an existing user. The type of a user cannot be changed
- [x] **LockUser**: changes the state of the user to `locked`
- [x] **UnlockUser**: changes the state of the user to `active` after it was locked
- [x] **ActivateUser**: changes the state of the user to `inactive`
- [x] **ReactivateUser**: changes the state of the user to `active` after it was locked
- [x]  the user data is filled according to the predefined user schema
- [x] As an admin I am able to change all fields that are defined as writeable by admin
- [x] As an end user I am not able to change fields that are only writeable by an admin
- [x] if user doesn't match the userschema, human readable error message is shown
- [x] I am able to change the user schema of a user, the schema is checked against the user information
   - [x] If only the schema changes, the current user information is checked for validity
   - [x] If the user schema changes and new user information is sent, the user schema is checked against the new user information
   - [x] User authenticators are checked against the new user schema
- [x] Detailed validation errors are returned to the user

## Todos

- [x] UpdateUser
- [x] LockUser
- [x] UnlockUser
- [x] DeactivateUser
- [x] ReactivateUser

### User Object
* **id**: unchangeble, unique identifier of the system
* **schema**: 
    * **type**: schema type of the user (e.g. human/machine)
    * **revision**: read only, revision of the schema the user was stored the last time (e.g. 4)
* **authenticators**: list of possibilites how a user can authenticate itself. there can be multiple authenticators of the same type
* **contact**: email and phone of the user if provided
* **state**: one of `Active`, `Inactive`, `Locked`, `Deleted`
* **data/profile/traits**: custom fields of a user based on the type (whats the best name for this?)

### PR DESCRIPTION
**Pull request by stebenz**
# Which Problems Are Solved

Users are not yet able to update their information an status in user API v3.

# How the Problems Are Solved

Add endpoints and functionality to update users and their status in user API v3.

# Additional Changes

Aggregate_type and event_types are updated with "userschema" to avoid conflicts with old events.

# Additional Context

closes #7898

### CODE REVIEW

File: internal/command/user_v3_test.go [hunk 1]
```diff
@@ -1101,3 +1103,2078 @@ func TestCommandSide_DeleteSchemaUser(t *testing.T) {
...
+				userID: "user1",
+			},
+			res: res{
+				err: func(err error) bool {
+					return errors.Is(err, zerrors.ThrowPermissionDenied(nil, "AUTHZ-HKJD33", "Errors.PermissionDenied"))
+				},
+			},
+		},
+		{
+			name: "lock user, self",
```
Comment (eliobischof) [inline internal/command/user_v3_test.go:1295]: Does this test expect that a user can lock himself?

File: internal/repository/user/schemauser/user.go [hunk 2]
```diff
@@ -8,10 +8,14 @@ import (
...
-	UpdatedType = eventPrefix + "updated"
-	DeletedType = eventPrefix + "deleted"
+	eventPrefix     = "schemauser."
+	CreatedType     = eventPrefix + "created"
+	UpdatedType     = eventPrefix + "updated"
+	DeletedType     = eventPrefix + "deleted"
+	LockedType      = eventPrefix + "locked"
+	UnlockedType    = eventPrefix + "unlocked"
+	DeactivatedType = eventPrefix + "deactivated"
+	ReactivatedType = eventPrefix + "reactivated"
```
Comment (eliobischof) [inline internal/repository/user/schemauser/user.go:18]:
```suggestion
	ActivatedType = eventPrefix + "activated"
```

File: internal/api/grpc/resources/user/v3alpha/integration_test/user_test.go [hunk 3]
```diff
@@ -221,6 +222,297 @@ func TestServer_CreateUser(t *testing.T) {
...
+				req.User.Data = unmarshalJSON(data)
+				req.User.SchemaId = gu.Ptr(schemaID)
+				return nil
+			},
+			req: &user.PatchUserRequest{
+				User: &user.PatchUser{},
+			},
+			res: res{
+				want: &resource_object.Details{
+					Changed: timestamppb.Now(),
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/api/grpc/resources/user/v3alpha/integration_test/user_test.go:397]: Yes, but as the timestamps are checked with the range this way it's more performance test that the call is not timeouted.

File: internal/integration/client.go [hunk 4]
```diff
@@ -784,3 +784,55 @@ func (i *Instance) CreateInviteCode(ctx context.Context, userID string) *user_v2
...
+func (i *Instance) LockSchemaUser(ctx context.Context, orgID string, userID string) *user_v3alpha.LockUserResponse {
+	var org *object_v3alpha.Organization
+	if orgID != "" {
+		org = &object_v3alpha.Organization{Property: &object_v3alpha.Organization_OrgId{OrgId: orgID}}
+	}
+	user, err := i.Client.UserV3Alpha.LockUser(ctx, &user_v3alpha.LockUserRequest{
+		Organization: org,
+		Id:           userID,
+	})
+	logging.OnError(err).Fatal("lock user")
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/integration/client.go:797]: Yes, as this is used for dependencies in the tests, they should never fail, same is used in other client functions.

File: internal/command/user_v3.go [hunk 5]
```diff
@@ -161,6 +161,239 @@ func (c *Commands) DeleteSchemaUser(ctx context.Context, id string) (*domain.Obj
...
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-pjJhge86ZV", "Errors.IDMissing")
+	}
+	writeModel, err := c.getSchemaUserExists(ctx, resourceOwner, id)
+	if err != nil {
+		return nil, err
+	}
+	if !writeModel.Exists() {
+		return nil, zerrors.ThrowNotFound(nil, "COMMAND-Ob6lR5iFTe", "Errors.User.NotFound")
+	}
+	if writeModel.State != domain.UserStateActive {
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/command/user_v3.go:359]: Locked is now a separate boolean besides the state.

File: internal/command/user_v3_test.go [hunk 6]
```diff
@@ -1101,3 +1103,2078 @@ func TestCommandSide_DeleteSchemaUser(t *testing.T) {
...
+				userID: "user1",
+			},
+			res: res{
+				err: func(err error) bool {
+					return errors.Is(err, zerrors.ThrowPermissionDenied(nil, "AUTHZ-HKJD33", "Errors.PermissionDenied"))
+				},
+			},
+		},
+		{
+			name: "unlock user, self",
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/command/user_v3_test.go]: Status changes have now a separate permission check, so self unlock is not possible anymore

File: internal/command/user_v3_test.go (hunk 1 above)
Comment (stebenz) [reply internal/command/user_v3_test.go]: Status changes have now a separate permission check, so self unlock is not possible anymore

File: internal/repository/user/schemauser/user.go (hunk 2 above)
Comment (stebenz) [reply internal/repository/user/schemauser/user.go]: Renamed to activated

File: proto/zitadel/resources/user/v3alpha/user_service.proto [hunk 7]
```diff
@@ -294,7 +294,7 @@ service ZITADELUsers {
   // The endpoint returns an error if the user is already in the state 'locked'.
   rpc LockUser (LockUserRequest) returns (LockUserResponse) {
     option (google.api.http) = {
-      post: "/resources/v3alpha/users/{user_id}/_lock"
+      post: "/resources/v3alpha/users/{id}/_lock"
```
[the remark this answers is not in the document]
Comment (stebenz) [reply proto/zitadel/resources/user/v3alpha/user_service.proto:297]: Lock and deactivate are two different options, which means they are not connected anymore, that's why I would avoid comments like this.

[the remark this answers is not in the document]
Comment (eliobischof) [reply proto/zitadel/resources/user/v3alpha/user_service.proto:297]: The idea is to help the user understand the difference between lock and deactivate.

File: internal/integration/client.go [hunk 8]
```diff
@@ -784,3 +784,55 @@ func (i *Instance) CreateInviteCode(ctx context.Context, userID string) *user_v2
...
+func (i *Instance) ActivateSchemaUser(ctx context.Context, orgID string, userID string) *user_v3alpha.ActivateUserResponse {
+	var org *object_v3alpha.Organization
+	if orgID != "" {
+		org = &object_v3alpha.Organization{Property: &object_v3alpha.Organization_OrgId{OrgId: orgID}}
+	}
+	user, err := i.Client.UserV3Alpha.ActivateUser(ctx, &user_v3alpha.ActivateUserRequest{
+		Organization: org,
+		Id:           userID,
+	})
+	logging.OnError(err).Fatal("reactivate user")
```
Comment (eliobischof) [inline internal/integration/client.go:836]: Do you log these errors intentionally?

**Review by eliobischof — CHANGES_REQUESTED** [review]

**Review by eliobischof — APPROVED** [review]

**Pull request merged**